telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
+    2. Writes benchmarks to ~/src/tools/perf/benchmarks/page_cycler.py if they
+       do not already exist
+    NOTE: You may need to clear out the end of
+    ~/src/tools/perf/benchmarks/page_cycler.py for a clean start!
+    :param trial_number: int for the number of this trial run
+    :param num_urls: int of the number of urls to run through
+    """
+    # Ensure benchmark_results.db exists
+    telemetry_record_path = \
+    '/home/jamshed/page_load_time/telemetry/temp/benchmark_results.db'
+    if not os.path.isfile(telemetry_record_path):
+        # Create file
+        tmp = {trial_number: {}}
+        pickle.dump(tmp, open(telemetry_record_path, 'wb'))
+    try:
+        curr_pickle = pickle.load(open(telemetry_record_path, 'rb'))
+    except IOError:
+        # Raise exception or rewrite benchmark_results.db?
+        tmp = {trial_number: {}}
+        pickle.dump(tmp, open(telemetry_record_path, 'wb'))
+
+    # Write benchmarks to page_cycler.py
+    page_cycler_path = '/home/jamshed/src/tools/perf/benchmarks/page_cycler.py'
+    benchmark_template = ("@benchmark.Enabled('android')\n"
+                          "class PageCyclerUrl{0}(_PageCycler):\n"
+                          "    page_set = page_sets.url{0}PageSet\n\n")
+
+    with open(page_cycler_path, 'r+b') as f:
+        line = f.readline()
+        already_written = False
+        while (line):
+            if '### END OF URLS ###' in line:
+                already_written = True
+                break
+            line = f.readline()
+        if not already_written:
+            # Need to write benchmarks to page_cycler.py, for now, just append
+            f.write('\n\n')
+            for i in range(num_urls):
+                f.write(benchmark_template.format(i))
+                f.write(benchmark_template.format('999999' + str(i)))
+            f.write('### END OF URLS ###')
+
+def run_benchmarks(num_urls):
+    """Runs the page_cycler benchmark for each url
+
+    Dumps data /temp/benchmark_results.db
+    :param num_urls: int of the number of urls to run through
+    """
+    path.append('/home/jamshed/src/tools/perf/')
+    cmd = ('sudo /home/jamshed/src/tools/perf/run_benchmark '
+           'page_cycler.PageCyclerUrl{0}')
+    for i in range(num_urls):
+        p = Popen(cmd.format(i), shell=True)
+        p.wait()  # We can only load 1 url on Chrome at a time
+        # Also run the modified version
+        p = Popen(cmd.format('999999' + str(i)), shell=True)
+        p.wait()  # We can only load 1 url on Chrome at a time
+
+def aggregate_benchmark_data():
+    """Merges data from /temp/benchmark_results.db with /temp/aggregate.db
+
+    Note: temp/aggregate.db looks like {'http://jamnoise.com':
+                                           {'cold_times':
+                                               {'trial1': [data],...}}}
+    """
+    benchmark_results = '/data/benchmark_results.db'
+    benchmark_result_path = \
```

Comment:
It would be good to have all of these hardcoded strings pulled out into global constants rather than hardcoded here, especially since they're specific to your machine.